DataFormats/L1TrackTrigger/interface/TTTrack_TrackWord.h

```diff
@@ -268,14 +323,23 @@ class TTTrack_TrackWord {
     return (up - bins.begin() - 1);
   }
 
-  double unpackSignedValue(unsigned int bits, unsigned int nBits, double lsb) const {
-    int isign = 1;
-    unsigned int digitized_maximum = (1 << nBits) - 1;
-    if (bits & (1 << (nBits - 1))) {  // check the sign
-      isign = -1;
-      bits = (1 << (nBits + 1)) - bits;  // if negative, flip everything for two's complement encoding
+  double undigitizeSignedValue(unsigned int twosValue, unsigned int nBits, double lsb) const {
+    // Check that none of the bits above the nBits-1 bit, in a range of [0, nBits-1], are set.
```

Comment:
I'm not sure why this check is needed. But can't it be done in a single line of code, eliminating the need for the countSetBits* functions? e.g.:
assert (twosValue >> (lsb + nbits) == 0);

Reply:
I think it should be `assert((twosValue >> nbits) == 0)`

Reply:
The check is now done with the line suggested by @gpetruc.